activity.cc

```diff
@@ -129,4 +132,99 @@ ActivityLog PolluteDataCache::GetActivityLog() {
   return alog;
 }
 
+// Activity: PolluteInstructionCache
+
+// The boost library (https://github.com/nelhage/rules_boost) is used to
+// create a large 2D array of functions with names -
+//
+// DummyFunc_0_0, DummyFunc_0_1, ..., DummyFunc_0_{POLLUTE_ICACHE_LOOP_SIZE-1},
+// ...
+// ...
+// ...
+// DummyFunc_{POLLUTE_ICACHE_LOOP_SIZE-1}_0, ..., DummyFunc_{POLLUTE_ICACHE_LOOP_SIZE-1}_{POLLUTE_ICACHE_LOOP_SIZE-1}.
+// Each of these functions returns an int.
+//
+// This 2D array was created because BOOST_PP_REPEAT() macro can
+// generate at most 1024 repetitions.
+//
+// The purpose of 'do_work' part of the function is only to increase
+// the size of the function.
+#define POLLUTE_ICACHE_LOOP_SIZE 255
+#define GENERATE_FUNC_INNER_LOOP(z, n, prefix)  \
+  int DummyFunc_ ## prefix ## _ ## n (bool do_work) {       \
+    if (do_work) {                \
+      int result = 0;             \
+      for (int i=0; i<n; i++) {   \
+        result += i*i;            \
+        result /= i+1;            \
+        result -= i/2;            \
+        result /= i*i;            \
+        result += result*i;       \
+        result /= result+1;       \
+        result -= result/2;       \
+        result *= result;         \
+      }                           \
+      return result;              \
+    } else                        \
+      return n;                   \
+  }
+#define GENERATE_FUNC_OUTER_LOOP(z, n, text) \
+  BOOST_PP_REPEAT(POLLUTE_ICACHE_LOOP_SIZE, GENERATE_FUNC_INNER_LOOP, n)
+
+// Get the pointers to the functions created above and store it in
+// func_ptr_array.
+#define GET_FUNC_PTR_INNER_LOOP(z, n, prefix) \
+  func_ptr_array_[(prefix * POLLUTE_ICACHE_LOOP_SIZE + n)] = &DummyFunc_ ## prefix ## _ ## n;
```

Comment:
use the \ at the end of line to make the macro fit in 80 columns

Reply:
Done